examples/build_tests/src/library.c

```diff
@@ -0,0 +1,18 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "nanoarrow/nanoarrow.h"
```

Comment:
I would add some minor bit of example code to both files. (no idea what thought ^^) This would make for a better example (e.g. custom namespace usage) and properly exercise linking.

Reply:
Yeah I can add a simple test to make sure the library links correctly. I started simple to make sure that all the headers were being included, but I agree that we need to verify that the library works too.

Reply:
I've added real examples now so I'm not sure that the version check is relevant anymore, but let me know what you're thinking.